lib/CArtHandler.h

```diff
@@ -249,8 +249,8 @@ class DLL_LINKAGE CArtHandler : public CHandlerBase<ArtifactID, Artifact, CArtif
 
 	boost::optional<std::vector<CArtifact*>&> listFromClass(CArtifact::EartClass artifactClass);
 
-	ArtifactPosition stringToSlot(std::string slotName);
-	CArtifact::EartClass stringToClass(std::string className);
+	static ArtifactPosition stringToSlot(std::string slotName); //Static is intentional here, because VLC singletones is not good

```

Comment:
@IvanSavenko i think than making this static is good because there is no need to use VLC singleton for just converting string to enum. Maybe there is an std:: method for this.

Reply:
StringToSlot became a constructor, stringToClass become static with TODO for reworking EartClass.